[BUG] [RUST] Rust server generator fails when security is defined

#### Bug Report Checklist

- [X] Have you provided a full/minimal spec to reproduce the issue?
- [X] Have you validated the input using an OpenAPI validator ([example](https://apitools.dev/swagger-parser/online/))?
- [X] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [X] Have you searched for related issues/PRs?
- [X] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

In any openapi 3.0.4 spec that defines a top level or path security option, the rust-server generator fails with a null pointer exception as follows:

[main] INFO  o.o.codegen.DefaultGenerator - Generating with dryRun=false
[main] INFO  o.o.codegen.DefaultGenerator - OpenAPI Generator: rust-server (server)
[main] INFO  o.o.codegen.DefaultGenerator - Generator 'rust-server' is considered stable.
[main] INFO  o.o.c.languages.RustServerCodegen - Environment variable RUST_POST_PROCESS_FILE not defined. rustfmt will be used by default. To choose a different tool, try 'export RUST_POST_PROCESS_FILE="/usr/local/bin/rustfmt"' (Linux/Mac)
[main] INFO  o.o.c.languages.RustServerCodegen - NOTE: To enable file post-processing, 'enablePostProcessFile' must be set to `true`  (--enable-post-process-file for CLI).
[main] INFO  o.o.codegen.TemplateManager - writing file /local/docs/GetVar.md
[main] INFO  o.o.codegen.TemplateManager - writing file /local/docs/SetVar.md
[main] INFO  o.o.codegen.TemplateManager - writing file /local/docs/Time.md
[main] WARN  o.o.c.languages.AbstractRustCodegen - ApplicationJson cannot be used as a model name. Renamed to ApplicationSlashJson
[main] WARN  o.o.c.languages.AbstractRustCodegen - ApplicationJson cannot be used as a model name. Renamed to ApplicationSlashJson
[main] WARN  o.o.c.languages.AbstractRustCodegen - ApplicationJson cannot be used as a model name. Renamed to ApplicationSlashJson
Exception in thread "main" java.lang.RuntimeException: Could not generate api file for 'Time'
        at org.openapitools.codegen.DefaultGenerator.generateApis(DefaultGenerator.java:823)
        at org.openapitools.codegen.DefaultGenerator.generate(DefaultGenerator.java:1306)
        at org.openapitools.codegen.cmd.Generate.execute(Generate.java:535)
        at org.openapitools.codegen.cmd.OpenApiGeneratorCommand.run(OpenApiGeneratorCommand.java:32)
        at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:66)
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because "s.isOAuth" is null
        at org.openapitools.codegen.languages.RustServerCodegen.postProcessOperationWithModels(RustServerCodegen.java:958)
        at org.openapitools.codegen.languages.RustServerCodegen.postProcessOperationsWithModels(RustServerCodegen.java:819)
        at org.openapitools.codegen.DefaultGenerator.processOperations(DefaultGenerator.java:1665)
        at org.openapitools.codegen.DefaultGenerator.generateApis(DefaultGenerator.java:695)
        ... 4 more

##### openapi-generator version

7.12.0-SNAPSHOT

##### OpenAPI declaration file content or url

openapi: 3.0.4
info:
  title: Swagger Minimal Test API
  description: |-
    This is a minimal test API to allow testing of features in the the openapi tools code generators.
  contact:
    email: SoftwareEngineer@place.com
  license:
    name: Apache 2.0
    url: http://www.apache.org/licenses/LICENSE-2.0.html
  version: 1.0.0
servers:
  - url: http://localhost/api/v1
tags:
  - name: Time
    description: Get the time with a simple request
  - name: Variable
    description: Test setting a remote variable
  - name: File
    description: Test file uploads
paths:
  /var:
    post:
      tags:
        - Variable
      summary: Set the variable
      description: Set the variable with a json value
      operationId: varPost
      requestBody:
        description: Set the remote variable
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/SetVar'
        required: true
      responses:
        '200':
          description: Successful operation
        '400':
          description: Invalid input
        '422':
          description: Validation exception
    get:
      tags:
        - Variable
      summary: Get the current value of the server variable
      description: Just a get request to return the current server variable value.
      operationId: getVar
      responses:
        '200':
          description: successful operation
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/GetVar'
        '400':
          description: Bad request

  /time:
    get:
      tags:
        - Time
      summary: Get the current system time
      description: Just a get request to return the current server time.
      operationId: getTime
      responses:
        '200':
          description: successful operation
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Time'
        '400':
          description: Bad request

components:
  schemas:

    GetVar:
      type: object
      properties:
        Value:
          type: string
          description: Value for this server variable.

    SetVar:
      type: object
      properties:
        Value:
          type: string
          description: Set a value for this server variable.

    Time:
      type: object
      properties:
        timestring:
          type: string

  securitySchemes:
    openId:
      type: openIdConnect
      openIdConnectUrl: https://172.16.33.9:8443/realms/test/.well-known/openid-configuration


security:
  - openId:
    - var_read
    - var_write
    - time

##### Generation Details

docker run --rm         -v $PWD:/local         openapitools/openapi-generator-cli generate         -i /local/MinimalTest.http.oidc.304.yaml         -g rust-server         -o /local/         --additional-properties=packageName=testapi,generateAliasAsModel=true

##### Steps to reproduce

Run the above generation on the above yaml.

##### Related issues/PRs

Can't find any

##### Suggest a fix

Maybe initialize the s.isOauth variable?

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java
@@ -5734,7 +5734,7 @@ public class DefaultCodegen implements CodegenConfig {
         cs.name = key;
         cs.description = securityScheme.getDescription();
         cs.type = securityScheme.getType().toString();
-        cs.isCode = cs.isPassword = cs.isApplication = cs.isImplicit = cs.isOpenId = false;
+        cs.isCode = cs.isPassword = cs.isApplication = cs.isImplicit = cs.isOpenId = cs.isOAuth = false;
         cs.isHttpSignature = false;
         cs.isBasicBasic = cs.isBasicBearer = false;
         cs.scheme = securityScheme.getScheme();

--- a/modules/openapi-generator/src/test/java/org/openapitools/codegen/DefaultCodegenTest.java
+++ b/modules/openapi-generator/src/test/java/org/openapitools/codegen/DefaultCodegenTest.java
@@ -5036,6 +5036,19 @@ public class DefaultCodegenTest {
         assertTrue(codegenOperation.queryParams.stream().allMatch(p -> p.queryIsJsonMimeType));
     }
 
+    @Test
+    public void testDefaultOauthIsNotNull() {
+        final DefaultCodegen codegen = new DefaultCodegen();
+        final OpenAPI openAPI = TestUtils.parseFlattenSpec("src/test/resources/3_1/issue_20662.yaml");
+        codegen.setOpenAPI(openAPI);
+        List<CodegenSecurity> securitySchemes = codegen.fromSecurity(openAPI.getComponents().getSecuritySchemes());
+        assertThat(securitySchemes.size()).isEqualTo(1);
+        CodegenSecurity openIdScheme = securitySchemes.get(0);
+        assertNotNull(openIdScheme.isOAuth);
+        assertFalse(openIdScheme.isOAuth);
+        assertTrue(openIdScheme.isOpenId);
+    }
+
     private List<String> getRequiredVars(CodegenModel model) {
         return getNames(model.getRequiredVars());
     }

--- a/modules/openapi-generator/src/test/resources/3_1/issue_20662.yaml
+++ b/modules/openapi-generator/src/test/resources/3_1/issue_20662.yaml
@@ -1,0 +1,22 @@
+openapi: 3.1.0
+info:
+  title: Security scheme test
+  version: 1.0.0
+paths:
+  /test:
+    get:
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: string
+components:
+  securitySchemes:
+    openId:
+      type: openIdConnect
+      openIdConnectUrl: https://localhost:8443
+security:
+  - openId:
+      - test

